src/common/run-post-step.ts

```diff
@@ -0,0 +1,58 @@
+import { getInput } from "@actions/core";
+import { context } from "@actions/github";
+import { createClient } from "@alwaysmeticulous/client";
+import { getInputFromEnv } from "./get-input-from-env";
+import { getOctokitOrFail } from "./octokit";
+
+export const runPostStep = async (): Promise<void> => {
+  const apiToken = getActionInput("api-token");
+  const githubToken = getActionInput("github-token");
+  const octokit = getOctokitOrFail(githubToken);
+  const workflow = await octokit.rest.actions.getWorkflowRun({
+    owner: context.repo.owner,
+    repo: context.repo.repo,
+    run_id: context.runId,
+  });
+
+  const values: Record<string, number> = {
+    "report_diffs_action.was_cancelled":
+      workflow.data.status === "cancelled" ? 1 : 0,
+  };
+
+  // This should always be true since the workflow is running, but we need to check it to make the compiler happy
+  if (workflow.data.run_started_at) {
+    const timeSinceStart =
+      new Date().getTime() - new Date(workflow.data.run_started_at).getTime();
+    values["report_diffs_action.job_duration_seconds"] = timeSinceStart / 1000;
```

Comment:
This is the duration of the whole job rather than just this step? Nice, that'll be useful to track.

Reply:
Exactly, so we can track the duration of their build with this.